src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PETypeParameterSymbol.cs

```diff
@@ -309,11 +309,22 @@ public override bool HasReferenceTypeConstraint
             }
         }
 
+        public override bool HasNotnullConstraint
+        {
+            get
+            {
+                return (_flags & (GenericParameterAttributes.NotNullableValueTypeConstraint | GenericParameterAttributes.ReferenceTypeConstraint)) == 0 &&
+                       ((PEModuleSymbol)this.ContainingModule).Module.HasNullableAttribute(_handle, out byte transformFlag, out _) &&
+                       transformFlag == NullableAnnotationExtensions.NotAnnotatedAttributeValue;
+            }
+        }
+
         internal override bool? IsNotNullableIfReferenceType
         {
             get
             {
```

Comment:
>{ [](start = 12, length = 1)

Could we do `if (HasNotnullConstraint) { return true; }`?

Reply:
> Could we do if (HasNotnullConstraint) { return true; }?

We could (as in other implementations of IsNotNullableIfReferenceType), but I decided against duplicating the logic around that and keep it in ```CalculateIsNotNullableIfReferenceType```. 

---
In reply to: [289953856](https://github.com/dotnet/roslyn/pull/36109#discussion_r289953856) [](ancestors = 289953856)